pkg/controller/autoscaler/metrics_client.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscalercontroller
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/expapi"
+
+	heapster "k8s.io/heapster/api/v1/types"
+)
+
+// An interface for getting metrics for pods.
+type MetricsClient interface {
+	// Gets average resource consumption for a list of pods.
+	GetConsumptionFor(api.ResourceName, string, []string) (*expapi.ResourceConsumption, error)
```

Comment:
@smarterclayton Refactored so that there is a namespacer and get takes resource name and selector. 

@lavalamp  Added parameter names. For now I would rather not move it to the client package as the interface/implementation is unstable and tailored only for autoscaler.